src/Resolver/Resolver.php

```diff
@@ -20,17 +20,24 @@ public function __construct($nameserver, ExecutorInterface $executor)
 
     public function resolve($domain)
     {
+        return $this->resolveAll($domain)->then(function ($addresses) {
+                return $addresses[array_rand($addresses)];
+            });
+    }
+
+    public function resolveAll($domain)
+    {
         $query = new Query($domain, Message::TYPE_A, Message::CLASS_IN, time());
         $that = $this;
 
         return $this->executor
             ->query($this->nameserver, $query)
             ->then(function (Message $response) use ($query, $that) {
-                return $that->extractAddress($query, $response);
+                return $that->extractAddresses($query, $response);
             });
     }
 
-    public function extractAddress(Query $query, Message $response)
+    public function extractAddresses(Query $query, Message $response)
```

Comment:
I agree here, while this method isn't *intended* to be used from the outside, it's still part of our (current) public API.